bundles/org.openhab.persistence.dynamodb/src/main/java/org/openhab/persistence/dynamodb/internal/DynamoDBPersistenceService.java

```diff
@@ -221,7 +221,17 @@ public void activate(final @Nullable BundleContext bundleContext, final Map<Stri
             writeBufferedDataFuture = scheduler.scheduleWithFixedDelay(new Runnable() {
                 @Override
                 public void run() {
-                    DynamoDBPersistenceService.this.flushBufferedData();
+                    try {
+                        DynamoDBPersistenceService.this.flushBufferedData();
+                    } catch (Exception e) {
```

Comment:
If you catch you should use `RuntimeException` while your changing code here you could change the use of Runnable into a Java 8 notation.

Reply:
I do want to catch checked exceptions as well in case they appear at some point.

I tried to explain this in the comments
```
                       // We want to catch all unexpected exceptions since all unhandled exceptions make
                        // ScheduledExecutorService halt the regular running of the task.
                        // It is better to print out the exception, and try again
                        // (on next cycle)
```

Reply:
But checked exceptions can only be thrown when a methods says it throws such an exception. So unless a method indicates it throws `Exception` you should get a compile error if you don't catch a checked exception, because the method run doesn't throw any exception. Only unchecked exceptions aka RuntimeExceptions can be thrown unexpected. Therefore it's good practice to catch only RuntimeException and any thrown checked exception. You won't miss any exceptions that way.

Reply:
I understand, currently the method is not declaring any checked exceptions.

In case it will have checked exceptions some day, I do want to handle them exactly the same way as runtime exceptions... By retrying again automatically after a while.

I generally agree that catching all exceptions is typically bad. However, in this case I do want it to avoid halting the regular schedule set by scheduleWithFixedDelay.

With Errors I consider it understandable that this regular cycle is broken, so those are not catched.

I hope this clarifies my intention.

Reply:
I have no problem with catching all exceptions, just with using `Exception` instead of `RuntimeException`. For example if you would get the checked exception `InteruptedException` you should probably stop the scheduler. So not all checked Exceptions should be handled the same, and if you catch the generic Exception you might miss that case. There is a reason these are checked exceptions. Hence not to use `Exception`, but use `RuntimeException`. This will have the effect in that it give you the means to let the scheduler keep running.

Reply:
Ah good point indeed with the interruption as example. And now I reread your comment on the Runnable.run -- indeed it does not allow checked exceptions and as such programmer will be "notified" and adjusting the exception catching logic is needed.

Agree now with your suggestion and modify the code accordingly. 👍